libp2p/protocols/pubsub/timedcache.nim

```diff
@@ -83,16 +102,24 @@ func put*[K](t: var TimedCache[K], k: K, now = Moment.now()): bool =
       if cur == t.tail:
         t.tail = node
 
-  t.entries[k] = node
+  t.entries.incl(node)
 
   previous.isSome()
 
 func contains*[K](t: TimedCache[K], k: K): bool =
-  k in t.entries
-
-func addedAt*[K](t: TimedCache[K], k: K): Moment =
-  t.entries.getOrDefault(k).addedAt
-
+  let tmp = TimedEntry[K](key: k)
+  tmp in t.entries
+
+func addedAt*[K](t: var TimedCache[K], k: K): Moment =
```

Comment:
What about returning Opt[Moment] and letting the caller decide what to do?

Reply:
this code matches the previous behavior of returning a default-initialized Moment - I don't see the need to change this in this PR.